src/deploy/NVA_build/noobaa_core.yaml

```diff
@@ -171,6 +177,15 @@ spec:
                   name: noobaa-create-sys-creds
                   key: password
                   optional: true
+            - name: OAUTH_SERVICE_HOST
```

Comment:
Can you clarify? 
This is an environment variable taken from a config map ser by our deployment script/operator.
The discovery of the auth server is done there and the hostname of the auth server  is passed to NooBaa using this env variable

Reply:
You need to handle `authorization_endpoint` and `token_endpoint` separately.

Reply:
@enj Out of curiosity, why would these endpoints be different?

Reply:
> @enj Out of curiosity, why would these endpoints be different?

I can interpret that question in two ways so I will try to answer both.  `authorization_endpoint` and `token_endpoint` are of course different from each other, so they must be handled distinctly.  But I do not think that is what you are actually asking.

Instead I believe you are asking "why would the values be different from what the built in OAuth server provides."  The reason for this is that we allow replacing the OAuth server.  You could theoretically wire multiple clusters to point to the same OAuth server.  That OAuth server would likely be KeyCloak, but it could technically be the integrated OAuth server from a different cluster.  The work to actually make KeyCloak work correctly in this case is still in the early stages, but the wiring to do it correctly in OpenShift was present in 3.11.  Thus the console, oc, OAuth proxy, etc all make no assumptions about the OAuth server.  They check the well-known endpoint (which I added in 3.4) and follow it to whatever the cluster is configured to use.

Reply:
It will be trivial to move the construction/discovery of the full auth endpoints to the deploy script/operator and pass both of them to the NooBaa pod. I have no objection in this regard.